src/client/components/pages/Courses/tableFields.tsx

```diff
@@ -143,6 +148,60 @@ export const formatMeetings = (
     id: courseId,
   } = course;
   const [modalVisible, setModalVisible] = useState(false);
+  /**
+   * Keeps track of whether the user has altered fields in the form to determine
+   * whether to show a confirmation dialog on modal close
+   */
+  const [
+    isChanged,
+    setIsChanged,
+  ] = useState(false);
+
+  const confirmMessage = "You have unsaved changes. Click 'OK' to disregard changes, or 'Cancel' to continue editing.";
+
+  useEffect(() => {
+    /**
+     * Checks to see if there are any unsaved changes in the modal When the user
+     * refreshes the page. If there are unsaved changes, the browser displays a
+     * warning message to confirm the page reload. If the user selects cancel, the
+     * user can continue making changes in the modal.
+     */
+    const onBeforeUnload = (event: Event) => {
+      if (!isChanged) return;
+      event.preventDefault();
+      // It's unclear whether TS will account for browser incompatibility here,
+      // so we use all three methods of setting the confirmation message,
+      // as detailed here:
+      // https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event
+      const legacyEvent = (event as unknown as LegacyEvent);
+      legacyEvent.returnValue = confirmMessage;
+      return confirmMessage;
+    };
+    window.addEventListener('beforeunload', onBeforeUnload);
+    return () => {
+      window.removeEventListener('beforeunload', onBeforeUnload);
+    };
+  },
```

Comment:
Just curious, did you consider using [the `history.block` method from React Router][rrhb]? I haven't explored it too deeply but it does seem to cover some of the edge cases mentioned in that mdn page.

[rrhb]: https://github.com/ReactTraining/history/blob/master/docs/blocking-transitions.md